src/murmur/ServerDB.cpp

```diff
@@ -1982,22 +1984,55 @@ int Server::readLastChannel(int id) {
 	if (id < 0)
 		return -1;
 
+	if (!Meta::mp.bRememberChan)
+		return -1;
+
 	TransactionHolder th;
 	QSqlQuery &query = *th.qsqQuery;
 
-	SQLPREP("SELECT `lastchannel` FROM `%1users` WHERE `server_id` = ? AND `user_id` = ?");
+	SQLPREP("SELECT `lastchannel`,`last_disconnect` FROM `%1users` WHERE `server_id` = ? AND `user_id` = ?");
 	query.addBindValue(iServerNum);
 	query.addBindValue(id);
 	SQLEXEC();
 
 	if (query.next()) {
 		int cid = query.value(0).toInt();
-		if (qhChannels.contains(cid))
-			return cid;
+		if (query.value(1).isNull()) {
+			return -1;
+		}
```

Comment:
I think we should turn this around: If this value is null, just return `cid` and only if it is set, check for the time. With this implementation I think there'd be an issue for a server that has just upgraded to the new version. The DB will not contain entries for `last_disconnect` yet and thus for the first time each registered user connects, the remember channel functionality is broken.

Besides: If this field is not set, only the forgetting part must break, not the remember channel as a whole...

This is especially true, if the server doesn't even have the forget feature active :point_up: 
```suggestion
		if (query.value(1).isNull()) {
			return qhChannels.contains(cid) ? cid : -1;
		}
```